### PR TITLE
Balances bitstmap

### PR DESCRIPTION
**Pull request by ainestal**
Get the balances from Bitstamp using the generic api.
Update Kraken and Poloniex to include the balances as described in the Exchange trait.
Introduce the Currencty enum to represent currencies. This allows us to reference specific currencies in each exchange.

For now it doesn't return an error if the credentials are wrong.

### CODE REVIEW

**Comment by hugues31** [comment]
Thanks! I don't see any problem with the implementation so I merge. I will implement balances for Kraken and Poloniex in a few days (maybe today in fact)

**Comment by ainestal** [comment]
There is one problem, the errors are not correctly handled. One of these errors happens when the credentials are not set.

I wanted to implement Kraken next, but feel free to go for it if you like

**Comment by hugues31** [comment]
We must add a new error like this:
```rust
BadCredentials {
            description("BadCredentials")
                display("The informations provided do not allow authentication.")
        }
```
You can implement Kraken, I will do Poloniex 😄

File: src/bitstamp/generic_api.rs [hunk 1]
```diff
@@ -99,4 +99,46 @@ impl ExchangeApi for BitstampApi {
...
+//                "xrpeur_fee" => "xrpeur_fee",
+//                "xrpbtc_fee" => "xrpbtc_fee",
+//                "fee" => "fee",
+                _ => continue,
+            };
+
+            let amount = row.1
+                .as_str()
+                .unwrap()
+                .parse::<f64>()?;
```
Comment (ainestal) [inline src/bitstamp/generic_api.rs]: I tried that and it didn't work. I have to review the code and include the errors

Comment (ainestal) [inline src/bitstamp/generic_api.rs]: I'll create a new PR fixing the errors. Thx for the help

**Pull request merged**